homeassistant/config_entries.py

```diff
@@ -124,6 +125,7 @@
 
 DISCOVERY_COOLDOWN = 1
 
+_DataT = TypeVar("_DataT", default=Any)
```

Comment:
Should there be a restriction on what is the bounding type for the data to be stored? From the change in the component seems that the data might be a dataclass. Are we expecting onrwould like to have some restriction interface to be implemented on the holding data type? It seems from the ConfigEntry responsibilities that this could be any class each component believes might work for them, but in some cases (like Event class) we want the holding data to be somehow castable to a dict or serializable to a json. Just raising this to make sure this is the intention before comiting and making further restrictions more complicated.

Reply:
This is not restricted (and should not be restricted IMHO).